README.md

```diff
@@ -1,6 +1,90 @@
-# trireg2-api
-A versioned OpenApi specification for the TriReg2 API. Kept in a separate repository to allow public access, and for
-clients to request changes.
+# TriReg2 OpenAPI specification
 
-## Release procedure
-The api is released using a [GitHub action](https://github.com/trifork/trireg2-api/actions/workflows/api-publish.yml)
\ No newline at end of file
+A versioned [OpenAPI v3](https://spec.openapis.org/oas/v3.1.0) specification for the TriReg2 API. Kept in a separate repository to allow public access, and for
+clients to request changes or suggestions via pull requests or adding issues to the GitHub issue tracker.
+
+The OpenAPI specification contains a user API for the new TriReg2 service.
+
+Later, it will also contain an admin API and import APIs towards the TriReg2 service.
+
+## Publish a new API
+The TriReg2 OpenAPI specification is published using a [GitHub action](https://github.com/trifork/trireg2-api/actions/workflows/api-publish.yml) which is triggered when creating a new release. 
+
+### Create a new release
+Navigate to the `Code` tab and find the `Releases` section, and press the `Draft a new release` button. Create a new tag, for example a tag named `v0.0.7`, and add a title for your release, for example `Release v0.0.7`.
+
+Check the check-box `This is a pre-release`, if this is a pre-release version, and then press the `Publish release` button.
+
+### Versioning
+
+(Work in progress..) 
+TODO
+
+## Download the API
+
+(Work in progress..) 
+Download the API as a zip file from Github Assets, which can be found here... TODO
+
+## Maven and Gradle artifacts
+
+(Work in progress..) 
+TODO 
+
+## Git Submodule dependency
+
+You can also add a dependency to this API using this repo as a [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) in your application.
+
+If you start from scratch and your application does not have this git submodule dependency yet, add it as a submodule with the command (in your application root folder):
+
+```
+git submodule add git@github.com:trifork/trireg2-api.git trireg2-api
+```
+
+If you have cloned a repository, that already has a dependency to this submodule, you need to initialize and update the `trireg2-api` submodule with the command (in your application root folder):
+
+```
+git submodule update --init --recursive
+```
+
+During development phase, if there are new changes to this API, update your submodule from remote with the command (in your application root folder):
+
+```
+git submodule update --recursive
+```
+
+When updating the git submodule, it will refer to the latest version released (i.e. the head of the main branch). If you want to use another version of the API, for example a work-in-progress version of the API, which has been pushed as a branch in the API repo, you can just check that branch of the submodule out in your application source code and use it.
+
```

Comment:
Can be done in the .gitmodules file:
```
[submodule "trireg2-api"]
	path = trireg2-api
	url = git@github.com:trifork/trireg2-api
        branch = main <-- Add branch name here
```
Then run 
```
git submodule update --remote
```
In some cases we need to manually add `branch = main` for CI tools to properly fetch the submodule. If not, it might claim it doesn't know which branch to pull.